tests/virt_autotest/qem/live_migration_src.pm

```diff
@@ -0,0 +1,213 @@
+# Copyright © 2021 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: Live migration test for source host
+#
+# - Get resource data and dst host IP address
+# - Collect guest data, name, version, arch, image path, reusability ...
+# - Generate profile from autoyast template based on guest data
+# - First stage guest install
+# - Second stage guest install
+# - Get IP address of Guest and web page content. installation complete
+# - Set up passwordless ssh authentication
+# - Execute migration test cases
+# - Heath check guest on the dts host after migration
+# - clean up
+#
+# Maintainer: Tony Yuan <tyuan@suse.com>
+
+package live_migration_src;
+use strict;
+use warnings;
+use base 'opensusebasetest';
+use testapi;
+use File::Copy 'copy';
+use File::Path 'make_path';
+use lockapi;
+use mmapi;
+use scheduler 'get_test_suite_data';
+
+my $vms            = {};
+my $vms_to_install = 0;
+my $images_path    = "/var/lib/libvirt/images";
+my $nfs_path       = get_var("NFS_PATH") ? get_var("NFS_PATH") : get_var('SUT_IP') . ":$images_path";
+my $guest          = get_required_var('GUEST');
```

Comment:
It seems that this var $guest is directly used as guest name in later code. Does it mean that with current code, it allows 1 guest for kvm and 2 guests(fv/pv) in one openqa job? Users may have the need to test multiple product guests in one job, can this be supported in code?

Reply:
yes, 1 guest for kvm and 2 guests(fv/pv) in one openqa job.   

Ok, I will have it support a list of guests.

Reply:
Yes, we have 6 kvm guests and 12 XEN.

Reply:
We have them in `%virt_autotest::common::guests`.

Reply:
> It seems that this var $guest is directly used as guest name in later code. Does it mean that with current code, it allows 1 guest for kvm and 2 guests(fv/pv) in one openqa job? Users may have the need to test multiple product guests in one job, can this be supported in code?

A quick test run with two guests (15sp1, sp3) installed and some other fixes http://10.67.183.130/tests/753